pkg/engine/mutation.go

```diff
@@ -15,6 +16,10 @@ import (
 	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
 )
 
+func isNil(i interface{}) bool {
+	return i == nil || reflect.ValueOf(i).IsNil()
+}
```

Comment:
FYI an interface with a nill value was passed. Casing the `i == nil` to return false.

Reply:
With this change the error is actually returned:
`Handler factory requires a client but a nil client was passed, likely due to a bug or unsupported operation` 

So i assume that the 'bug' case in the statement wasn't caught.